pkg/scheduler/framework/runtime/framework.go

```diff
@@ -902,24 +902,24 @@ func (f *frameworkImpl) RunScorePlugins(ctx context.Context, state *framework.Cy
 	defer func() {
 		metrics.FrameworkExtensionPointDuration.WithLabelValues(score, status.Code().String(), f.profileName).Observe(metrics.SinceInSeconds(startTime))
 	}()
-	pluginToNodeScores := make(framework.PluginToNodeScores, len(f.scorePlugins))
+	pluginToNodeScores := make(map[framework.ScorePlugin]framework.NodeScoreList, len(f.scorePlugins))
 	for _, pl := range f.scorePlugins {
-		pluginToNodeScores[pl.Name()] = make(framework.NodeScoreList, len(nodes))
+		pluginToNodeScores[pl] = make(framework.NodeScoreList, len(nodes))
 	}
 	ctx, cancel := context.WithCancel(ctx)
 	errCh := parallelize.NewErrorChannel()
 
 	// Run Score method for each node in parallel.
 	f.Parallelizer().Until(ctx, len(nodes), func(index int) {
-		for _, pl := range f.scorePlugins {
+		for pl, nl := range pluginToNodeScores {
```

Comment:
This PR violates this constraint. 

https://github.com/kubernetes/kubernetes/blob/a38bb7ed811a18cabe73974afea86f6da34d142b/staging/src/k8s.io/kube-scheduler/config/v1/types.go#L222
```
	// These are called after default plugins and in the same order specified here.
```